diff --git xorgproto-clean/include/X11/Xfuncs.h xorgproto-workdir/include/X11/Xfuncs.h
index b23c283..82c856c 100644
--- xorgproto-clean/include/X11/Xfuncs.h
+++ xorgproto-workdir/include/X11/Xfuncs.h
@@ -44,7 +44,7 @@ void bcopy();
 #    define bcmp(b1,b2,len) memcmp(b1, b2, len)
 #   else
 #    include <string.h>
-#    if defined(__SCO__) || defined(__sun) || defined(__UNIXWARE__) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__)
+#    if defined(__SCO__) || defined(__sun) || defined(__UNIXWARE__) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__) || defined(__astral__)
 #     include <strings.h>
 #    endif
 #    define _XFUNCS_H_INCLUDED_STRING_H
diff --git xorgproto-clean/include/X11/Xos.h xorgproto-workdir/include/X11/Xos.h
index 28dfc67..f46490a 100644
--- xorgproto-clean/include/X11/Xos.h
+++ xorgproto-workdir/include/X11/Xos.h
@@ -60,7 +60,7 @@ in this Software without prior written authorization from The Open Group.
  */
 
 # include <string.h>
-# if defined(__SCO__) || defined(__UNIXWARE__) || defined(__sun) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__)
+# if defined(__SCO__) || defined(__UNIXWARE__) || defined(__sun) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__) || defined(__astral__)
 #  include <strings.h>
 # else
 #  ifndef index
