diff --git git-clean/configure.ac git-workdir/configure.ac
index d1a96da..05aed44 100644
--- git-clean/configure.ac
+++ git-workdir/configure.ac
@@ -755,7 +755,7 @@ GIT_CONF_SUBST([LIBC_CONTAINS_LIBINTL])
 # Define NO_GETTEXT if you don't want Git output to be translated.
 # A translated Git requires GNU libintl or another gettext implementation
 AC_CHECK_HEADER([libintl.h],
-[NO_GETTEXT=],
+[NO_GETTEXT=YesPlease],
 [NO_GETTEXT=YesPlease])
 GIT_CONF_SUBST([NO_GETTEXT])
 
