diff --git bash-clean/externs.h bash-workdir/externs.h
index 47b80f6..063912f 100644
--- bash-clean/externs.h
+++ bash-workdir/externs.h
@@ -308,7 +308,7 @@ extern int getdtablesize (void);
 #endif /* !HAVE_GETDTABLESIZE */
 
 #if !defined (HAVE_GETHOSTNAME)
-extern int gethostname (char *, int);
+extern int gethostname (char *, size_t);
 #endif /* !HAVE_GETHOSTNAME */
 
 extern int getmaxgroups (void);
diff --git bash-clean/subst.c bash-workdir/subst.c
index e9d3e75..f787665 100644
--- bash-clean/subst.c
+++ bash-workdir/subst.c
@@ -32,6 +32,7 @@
 #endif
 #include <signal.h>
 #include <errno.h>
+#include <limits.h>
 
 #if defined (HAVE_UNISTD_H)
 #  include <unistd.h>
diff --git bash-clean/support/config.sub bash-workdir/support/config.sub
index 4aaae46..dfefff8 100644
--- bash-clean/support/config.sub
+++ bash-workdir/support/config.sub
@@ -1974,6 +1974,7 @@ case $os in
 	| aos* \
 	| aros* \
 	| atheos* \
+	| astral* \
 	| auroraux* \
 	| aux* \
 	| beos* \
diff --git bash-clean/support/shobj-conf bash-workdir/support/shobj-conf
index c3df351..0576089 100755
--- bash-clean/support/shobj-conf
+++ bash-workdir/support/shobj-conf
@@ -79,6 +79,10 @@ nsk-cc-tandem|nsk-c99-tandem)
 	SHOBJ_LDFLAGS='-shared -bglobalized -unres_symbols ignore'
 	;;
 
+astral*-*gcc*)
+	SHOBJ_CFLAGS=-fpic
+	;;
+
 sunos4*-*gcc*)
 	SHOBJ_CFLAGS=-fpic
 	SHOBJ_LD=/usr/bin/ld
