diff -ruN pfetch-workdir/pfetch pfetch/pfetch
--- pfetch-workdir/pfetch	2022-10-10 17:04:55.933115927 -0300
+++ pfetch/pfetch	2022-10-10 17:09:10.889780542 -0300
@@ -956,9 +956,9 @@
     #       allows indentation to continue naturally despite
     #       the use of '<<-EOF'.
     case ${1:-${PF_ASCII:-${distro:-$os}}} in
-            ([Aa]stral*)
+    	([Aa]stral*)
             read_ascii 2 <<-EOF
-                ${c7}                      .::.
+	${c7}                      .::.
                    .:'  .:
     *    ${c4}MMM8&&${c7}.::'  .:'
        ${c4}MMMM${c7}.::'${c4}&&${c7}  .:'
@@ -970,8 +970,8 @@
 ${c7}:'  .:'
 ${c7}'::'                *
 
-                EOF
-
+EOF
+	;;
         [Aa]lpine*)
             read_ascii 4 <<-EOF
 				${c4}   /\\ /\\
