diff -ruN readline-clean/support/config.sub readline/support/config.sub
--- readline-clean/support/config.sub	2022-10-14 16:21:57.553229749 -0300
+++ readline/support/config.sub	2022-10-14 16:22:12.479896273 -0300
@@ -1723,7 +1723,7 @@
 	# Now accept the basic system types.
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
-	gnu* | android* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	gnu* | android* | astral* |bsd* | mach* | minix* | genix* | ultrix* | irix* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]* \
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* |  plan9* | psp* | sim* | xray* | os68k* | v88r* \
