diff --git libxslt-clean/configure.ac libxslt-workdir/configure.ac
index 8a00c25..0c0f3b2 100644
--- libxslt-clean/configure.ac
+++ libxslt-workdir/configure.ac
@@ -67,7 +67,7 @@ AC_SUBST(LIBEXSLT_VERSION_INFO)
 AC_SUBST(LIBEXSLT_VERSION_NUMBER)
 AC_SUBST(LIBEXSLT_VERSION_EXTRA)
 
-AM_INIT_AUTOMAKE([1.16.3 foreign no-dist-gzip dist-xz])
+AM_INIT_AUTOMAKE([1.15.1 foreign no-dist-gzip dist-xz])
 AM_MAINTAINER_MODE([enable])
 AM_SILENT_RULES([yes])
 
