diff --git libgpg-error-clean/configure.ac libgpg-error-workdir/configure.ac
index e8a8b6d8..263b75d9 100644
--- libgpg-error-clean/configure.ac
+++ libgpg-error-workdir/configure.ac
@@ -604,7 +604,7 @@ if test x"$gl_use_threads" = xno; then
   AC_MSG_NOTICE([generated src/lock-obj-pub.native.h for $host])
 elif test x$cross_compiling = xyes; then
   case $host in
-    *-*-linux-gnu* | *-*-linux-musl*)
+    *-*-linux-gnu* | *-*-linux-musl* | *-*-astral*)
     AC_CHECK_TOOL(OBJDUMP, [objdump])
     if test -n "$OBJDUMP"; then
       lock_obj_h_generated=yes
