--- a/src/fcstat.c
+++ b/src/fcstat.c
@@ -375,1 +375,1 @@
-#  elif defined(__linux__) || defined(__EMSCRIPTEN__)
+#  elif defined(__linux__) || defined(__EMSCRIPTEN__) || defined(__astral__)
--- a/configure.ac
+++ b/configure.ac
@@ -26,1 +26,1 @@
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
@@ -195,1 +195,0 @@
-AC_CHECK_INCLUDES_DEFAULT
@@ -208,1 +208,0 @@
-AX_FUNC_SNPRINTF
--- a/git.mk
+++ b/git.mk
@@ -235,1 +235,0 @@
-all: $(srcdir)/.gitignore gitignore-recurse-maybe
