diff -ruN mpc-workdir/build-aux/config.sub mpc/build-aux/config.sub
--- mpc-workdir/build-aux/config.sub	2022-10-10 16:41:29.463127241 -0300
+++ mpc/build-aux/config.sub	2022-10-10 16:42:52.139793243 -0300
@@ -1338,7 +1338,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
 	# sysv* is not here because it comes later, after sysvr4.
-	gnu* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	gnu* | astral* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]*\
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* | kopensolaris* | plan9* \
