diff --git flac-clean/m4/ogg.m4 flac-workdir/m4/ogg.m4
index 3b1a28a..43aa848 100644
--- flac-clean/m4/ogg.m4
+++ flac-workdir/m4/ogg.m4
@@ -14,30 +14,10 @@ AC_ARG_WITH(ogg-libraries,AS_HELP_STRING([--with-ogg-libraries=DIR],[Directory w
 AC_ARG_WITH(ogg-includes,AS_HELP_STRING([--with-ogg-includes=DIR],[Directory where libogg header files are installed (optional)]), ogg_includes="$withval", ogg_includes="")
 AC_ARG_ENABLE(oggtest,AS_HELP_STRING([--disable-oggtest],[Do not try to compile and run a test Ogg program]),, enable_oggtest=yes)
 
-  if test "x$ogg_libraries" != "x" ; then
-    OGG_LIBS="-L$ogg_libraries"
-  elif test "x$ogg_prefix" = "xno" || test "x$ogg_prefix" = "xyes" ; then
-    OGG_LIBS=""
-  elif test "x$ogg_prefix" != "x" ; then
-    OGG_LIBS="-L$ogg_prefix/lib"
-  elif test "x$prefix" != "xNONE" ; then
-    OGG_LIBS="-L$prefix/lib"
-  fi
-
   if test "x$ogg_prefix" != "xno" ; then
     OGG_LIBS="$OGG_LIBS -logg"
   fi
 
-  if test "x$ogg_includes" != "x" ; then
-    OGG_CFLAGS="-I$ogg_includes"
-  elif test "x$ogg_prefix" = "xno" || test "x$ogg_prefix" = "xyes" ; then
-    OGG_CFLAGS=""
-  elif test "x$ogg_prefix" != "x" ; then
-    OGG_CFLAGS="-I$ogg_prefix/include"
-  elif test "x$prefix" != "xNONE"; then
-    OGG_CFLAGS="-I$prefix/include"
-  fi
-
   AC_MSG_CHECKING(for Ogg)
   if test "x$ogg_prefix" = "xno" ; then
     no_ogg="disabled"
