diff --git libffi-clean/src/closures.c libffi-workdir/src/closures.c
index f7bead6..5a41ff1 100644
--- libffi-clean/src/closures.c
+++ libffi-workdir/src/closures.c
@@ -27,7 +27,7 @@
    DEALINGS IN THE SOFTWARE.
    ----------------------------------------------------------------------- */
 
-#if defined __linux__ && !defined _GNU_SOURCE
+#if (defined __linux__ || defined __astral__) && !defined _GNU_SOURCE
 #define _GNU_SOURCE 1
 #endif
 
@@ -139,6 +139,9 @@ ffi_tramp_is_present (__attribute__((unused)) void *ptr)
    executable memory. */
 #  define FFI_MMAP_EXEC_WRIT 1
 # endif
+# if defined(__astral__)
+#  define FFI_MMAP_EXEC_WRIT 1
+# endif
 #endif
 
 #if FFI_MMAP_EXEC_WRIT && !defined FFI_MMAP_EXEC_SELINUX
